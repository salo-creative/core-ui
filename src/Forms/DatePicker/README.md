# DatePicker

This component is a Single Date Picker based on react-dates from AirBnB. It allows the user to select a date from a calendar pop-up and returns a moment object of that selected date for use in the app. The user can also set a minimum and maximum range for the date picker by giving it a string in DD-MM-YYYY format.

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import DatePicker from '@salo/core-ui/Forms/DatePicker';
```

Implement as follows

```javascript
<DatePicker
  onDateChange={date => this.setState({ date })}
  onFocusChange={({focused}) => this.setState({ focused })}
  label="DatePicker Label"
  helperText="Helper Text which appears below the input."
  name="Component's Name"
  dateRangeMin="04-07-1776"
  dateRangeMax="01-05-2025"
/>
```