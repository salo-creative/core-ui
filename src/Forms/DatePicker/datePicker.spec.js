import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import DatePicker from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <DatePicker />
  );
  expect(container.firstChild).toMatchSnapshot();
});