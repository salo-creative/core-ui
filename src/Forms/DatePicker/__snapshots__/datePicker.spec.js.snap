// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c0 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 2rem;
}

.c0 .PresetDateRangePicker_panel {
  padding: 0 22px 11px;
}

.c0 .PresetDateRangePicker_button {
  position: relative;
  height: 100%;
  text-align: center;
  background: 0 0;
  border: 2px solid #00a699;
  color: #00a699;
  padding: 4px 12px;
  margin-right: 8px;
  font: inherit;
  font-weight: 700;
  line-height: normal;
  overflow: visible;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  cursor: pointer;
}

.c0 .PresetDateRangePicker_button:active {
  outline: 0;
}

.c0 .PresetDateRangePicker_button__selected {
  color: #fff;
  background: #00a699;
}

.c0 .SingleDatePickerInput {
  display: inline-block;
  background-color: #fff;
}

.c0 .SingleDatePickerInput__withBorder {
  border-radius: 2px;
  border: 1px solid #dbdbdb;
}

.c0 .SingleDatePickerInput__rtl {
  direction: rtl;
}

.c0 .SingleDatePickerInput__disabled {
  background-color: #f2f2f2;
}

.c0 .SingleDatePickerInput__block {
  display: block;
}

.c0 .SingleDatePickerInput__showClearDate {
  padding-right: 30px;
}

.c0 .SingleDatePickerInput_clearDate {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  padding: 10px;
  margin: 0 10px 0 5px;
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 .SingleDatePickerInput_clearDate__default:focus,
.c0 .SingleDatePickerInput_clearDate__default:hover {
  background: #dbdbdb;
  border-radius: 50%;
}

.c0 .SingleDatePickerInput_clearDate__small {
  padding: 6px;
}

.c0 .SingleDatePickerInput_clearDate__hide {
  visibility: hidden;
}

.c0 .SingleDatePickerInput_clearDate_svg {
  fill: #82888a;
  height: 12px;
  width: 15px;
  vertical-align: middle;
}

.c0 .SingleDatePickerInput_clearDate_svg__small {
  height: 9px;
}

.c0 .SingleDatePickerInput_calendarIcon {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  padding: 10px;
  margin: 0 5px 0 10px;
}

.c0 .SingleDatePickerInput_calendarIcon_svg {
  fill: #82888a;
  height: 15px;
  width: 14px;
  vertical-align: middle;
}

.c0 .SingleDatePicker {
  position: relative;
  display: inline-block;
}

.c0 .SingleDatePicker__block {
  display: block;
}

.c0 .SingleDatePicker_picker {
  z-index: 1;
  background-color: #fff;
  position: absolute;
}

.c0 .SingleDatePicker_picker__rtl {
  direction: rtl;
}

.c0 .SingleDatePicker_picker__directionLeft {
  left: 0;
}

.c0 .SingleDatePicker_picker__directionRight {
  right: 0;
}

.c0 .SingleDatePicker_picker__portal {
  background-color: rgba(0,0,0,.3);
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.c0 .SingleDatePicker_picker__fullScreenPortal {
  background-color: #fff;
}

.c0 .SingleDatePicker_closeButton {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  padding: 15px;
  z-index: 2;
}

.c0 .SingleDatePicker_closeButton:focus,
.c0 .SingleDatePicker_closeButton:hover {
  color: darken(#cacccd,10%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .SingleDatePicker_closeButton_svg {
  height: 15px;
  width: 15px;
  fill: #cacccd;
}

.c0 .DayPickerKeyboardShortcuts_buttonReset {
  background: 0 0;
  border: 0;
  border-radius: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  cursor: pointer;
  font-size: 14px;
}

.c0 .DayPickerKeyboardShortcuts_buttonReset:active {
  outline: 0;
}

.c0 .DayPickerKeyboardShortcuts_show {
  width: 33px;
  height: 26px;
  position: absolute;
  z-index: 2;
}

.c0 .DayPickerKeyboardShortcuts_show::before {
  content: "";
  display: block;
  position: absolute;
}

.c0 .DayPickerKeyboardShortcuts_show__bottomRight {
  bottom: 0;
  right: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__bottomRight::before {
  border-top: 26px solid transparent;
  border-right: 33px solid #00a699;
  bottom: 0;
  right: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__bottomRight:hover::before {
  border-right: 33px solid #008489;
}

.c0 .DayPickerKeyboardShortcuts_show__topRight {
  top: 0;
  right: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__topRight::before {
  border-bottom: 26px solid transparent;
  border-right: 33px solid #00a699;
  top: 0;
  right: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__topRight:hover::before {
  border-right: 33px solid #008489;
}

.c0 .DayPickerKeyboardShortcuts_show__topLeft {
  top: 0;
  left: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__topLeft::before {
  border-bottom: 26px solid transparent;
  border-left: 33px solid #00a699;
  top: 0;
  left: 0;
}

.c0 .DayPickerKeyboardShortcuts_show__topLeft:hover::before {
  border-left: 33px solid #008489;
}

.c0 .DayPickerKeyboardShortcuts_showSpan {
  color: #fff;
  position: absolute;
}

.c0 .DayPickerKeyboardShortcuts_showSpan__bottomRight {
  bottom: 0;
  right: 5px;
}

.c0 .DayPickerKeyboardShortcuts_showSpan__topRight {
  top: 1px;
  right: 5px;
}

.c0 .DayPickerKeyboardShortcuts_showSpan__topLeft {
  top: 1px;
  left: 5px;
}

.c0 .DayPickerKeyboardShortcuts_panel {
  overflow: auto;
  background: #fff;
  border: 1px solid #dbdbdb;
  border-radius: 2px;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 2;
  padding: 22px;
  margin: 33px;
  text-align: left;
}

.c0 .DayPickerKeyboardShortcuts_title {
  font-size: 16px;
  font-weight: 700;
  margin: 0;
}

.c0 .DayPickerKeyboardShortcuts_list {
  list-style: none;
  padding: 0;
  font-size: 14px;
}

.c0 .DayPickerKeyboardShortcuts_close {
  position: absolute;
  right: 22px;
  top: 22px;
  z-index: 2;
}

.c0 .DayPickerKeyboardShortcuts_close:active {
  outline: 0;
}

.c0 .DayPickerKeyboardShortcuts_closeSvg {
  height: 15px;
  width: 15px;
  fill: #cacccd;
}

.c0 .DayPickerKeyboardShortcuts_closeSvg:focus,
.c0 .DayPickerKeyboardShortcuts_closeSvg:hover {
  fill: #82888a;
}

.c0 .CalendarDay {
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 14px;
  text-align: center;
}

.c0 .CalendarDay:active {
  outline: 0;
}

.c0 .CalendarDay__defaultCursor {
  cursor: default;
}

.c0 .CalendarDay__default {
  border: 1px solid #e4e7e7;
  color: #484848;
  background: #fff;
}

.c0 .CalendarDay__default:hover {
  background: #e4e7e7;
  border: 1px solid #e4e7e7;
  color: inherit;
}

.c0 .CalendarDay__hovered_offset {
  background: #f4f5f5;
  border: 1px double #e4e7e7;
  color: inherit;
}

.c0 .CalendarDay__outside {
  border: 0;
  background: #fff;
  color: #484848;
}

.c0 .CalendarDay__outside:hover {
  border: 0;
}

.c0 .CalendarDay__blocked_minimum_nights {
  background: #fff;
  border: 1px solid #eceeee;
  color: #cacccd;
}

.c0 .CalendarDay__blocked_minimum_nights:active,
.c0 .CalendarDay__blocked_minimum_nights:hover {
  background: #fff;
  color: #cacccd;
}

.c0 .CalendarDay__highlighted_calendar {
  background: #ffe8bc;
  color: #484848;
}

.c0 .CalendarDay__highlighted_calendar:active,
.c0 .CalendarDay__highlighted_calendar:hover {
  background: #ffce71;
  color: #484848;
}

.c0 .CalendarDay__selected_span {
  background: #66e2da;
  border: 1px double #33dacd;
  color: #fff;
}

.c0 .CalendarDay__selected_span:active,
.c0 .CalendarDay__selected_span:hover {
  background: #33dacd;
  border: 1px double #33dacd;
  color: #fff;
}

.c0 .CalendarDay__selected,
.c0 .CalendarDay__selected:active,
.c0 .CalendarDay__selected:hover {
  background: #00a699;
  border: 1px double #00a699;
  color: #fff;
}

.c0 .CalendarDay__hovered_span,
.c0 .CalendarDay__hovered_span:hover {
  background: #b2f1ec;
  border: 1px double #80e8e0;
  color: #007a87;
}

.c0 .CalendarDay__hovered_span:active {
  background: #80e8e0;
  border: 1px double #80e8e0;
  color: #007a87;
}

.c0 .CalendarDay__blocked_calendar,
.c0 .CalendarDay__blocked_calendar:active,
.c0 .CalendarDay__blocked_calendar:hover {
  background: #cacccd;
  border: 1px solid #cacccd;
  color: #82888a;
}

.c0 .CalendarDay__blocked_out_of_range,
.c0 .CalendarDay__blocked_out_of_range:active,
.c0 .CalendarDay__blocked_out_of_range:hover {
  background: #fff;
  border: 1px solid #e4e7e7;
  color: #cacccd;
}

.c0 .CalendarDay__hovered_start_first_possible_end {
  background: #eceeee;
  border: 1px double #eceeee;
}

.c0 .CalendarDay__hovered_start_blocked_min_nights {
  background: #eceeee;
  border: 1px double #e4e7e7;
}

.c0 .CalendarMonth {
  background: #fff;
  text-align: center;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .CalendarMonth_table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 .CalendarMonth_verticalSpacing {
  border-collapse: separate;
}

.c0 .CalendarMonth_caption {
  color: #484848;
  font-size: 18px;
  text-align: center;
  padding-top: 22px;
  padding-bottom: 37px;
  caption-side: initial;
}

.c0 .CalendarMonth_caption__verticalScrollable {
  padding-top: 12px;
  padding-bottom: 7px;
}

.c0 .CalendarMonthGrid {
  background: #fff;
  text-align: left;
  z-index: 0;
}

.c0 .CalendarMonthGrid__animating {
  z-index: 1;
}

.c0 .CalendarMonthGrid__horizontal {
  position: absolute;
  left: 9px;
}

.c0 .CalendarMonthGrid__vertical,
.c0 .CalendarMonthGrid__vertical_scrollable {
  margin: 0 auto;
}

.c0 .CalendarMonthGrid_month__horizontal {
  display: inline-block;
  vertical-align: top;
  min-height: 100%;
}

.c0 .CalendarMonthGrid_month__hideForAnimation {
  position: absolute;
  z-index: -1;
  opacity: 0;
  pointer-events: none;
}

.c0 .CalendarMonthGrid_month__hidden {
  visibility: hidden;
}

.c0 .DayPickerNavigation {
  position: relative;
  z-index: 2;
}

.c0 .DayPickerNavigation__horizontal {
  height: 0;
}

.c0 .DayPickerNavigation__verticalDefault {
  position: absolute;
  width: 100%;
  height: 52px;
  bottom: 0;
  left: 0;
}

.c0 .DayPickerNavigation__verticalScrollableDefault {
  position: relative;
}

.c0 .DayPickerNavigation__bottom {
  height: auto;
}

.c0 .DayPickerNavigation__bottomDefault {
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-justify-content: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .DayPickerNavigation_button {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 0;
  padding: 0;
  margin: 0;
}

.c0 .DayPickerNavigation_button__default {
  border: 1px solid #e4e7e7;
  background-color: #fff;
  color: #757575;
}

.c0 .DayPickerNavigation_button__default:focus,
.c0 .DayPickerNavigation_button__default:hover {
  border: 1px solid #c4c4c4;
}

.c0 .DayPickerNavigation_button__default:active {
  background: #f2f2f2;
}

.c0 .DayPickerNavigation_button__disabled {
  cursor: default;
  border: 1px solid #f2f2f2;
}

.c0 .DayPickerNavigation_button__disabled:focus,
.c0 .DayPickerNavigation_button__disabled:hover {
  border: 1px solid #f2f2f2;
}

.c0 .DayPickerNavigation_button__disabled:active {
  background: 0 0;
}

.c0 .DayPickerNavigation_button__horizontalDefault {
  position: absolute;
  top: 18px;
  line-height: .78;
  border-radius: 3px;
  padding: 6px 9px;
}

.c0 .DayPickerNavigation_bottomButton__horizontalDefault {
  position: static;
  margin: -10px 22px 30px;
}

.c0 .DayPickerNavigation_leftButton__horizontalDefault {
  left: 22px;
}

.c0 .DayPickerNavigation_rightButton__horizontalDefault {
  right: 22px;
}

.c0 .DayPickerNavigation_button__verticalDefault {
  padding: 5px;
  background: #fff;
  box-shadow: 0 0 5px 2px rgba(0,0,0,.1);
  position: relative;
  display: inline-block;
  text-align: center;
  height: 100%;
  width: 50%;
}

.c0 .DayPickerNavigation_nextButton__verticalDefault {
  border-left: 0;
}

.c0 .DayPickerNavigation_nextButton__verticalScrollableDefault {
  width: 100%;
}

.c0 .DayPickerNavigation_svg__horizontal {
  height: 19px;
  width: 19px;
  fill: #82888a;
  display: block;
}

.c0 .DayPickerNavigation_svg__vertical {
  height: 42px;
  width: 42px;
  fill: #484848;
}

.c0 .DayPickerNavigation_svg__disabled {
  fill: #f2f2f2;
}

.c0 .DayPicker {
  background: #fff;
  position: relative;
  text-align: left;
}

.c0 .DayPicker__horizontal {
  background: #fff;
}

.c0 .DayPicker__verticalScrollable {
  height: 100%;
}

.c0 .DayPicker__hidden {
  visibility: hidden;
}

.c0 .DayPicker__withBorder {
  box-shadow: 0 2px 6px rgba(0,0,0,.05),0 0 0 1px rgba(0,0,0,.07);
  border-radius: 3px;
}

.c0 .DayPicker_portal__horizontal {
  box-shadow: none;
  position: absolute;
  left: 50%;
  top: 50%;
}

.c0 .DayPicker_portal__vertical {
  position: initial;
}

.c0 .DayPicker_focusRegion {
  outline: 0;
}

.c0 .DayPicker_calendarInfo__horizontal,
.c0 .DayPicker_wrapper__horizontal {
  display: inline-block;
  vertical-align: top;
}

.c0 .DayPicker_weekHeaders {
  position: relative;
}

.c0 .DayPicker_weekHeaders__horizontal {
  margin-left: 9px;
}

.c0 .DayPicker_weekHeader {
  color: #757575;
  position: absolute;
  top: 62px;
  z-index: 2;
  text-align: left;
}

.c0 .DayPicker_weekHeader__vertical {
  left: 50%;
}

.c0 .DayPicker_weekHeader__verticalScrollable {
  top: 0;
  display: table-row;
  border-bottom: 1px solid #dbdbdb;
  background: #fff;
  margin-left: 0;
  left: 0;
  width: 100%;
  text-align: center;
}

.c0 .DayPicker_weekHeader_ul {
  list-style: none;
  margin: 1px 0;
  padding-left: 0;
  padding-right: 0;
  font-size: 14px;
}

.c0 .DayPicker_weekHeader_li {
  display: inline-block;
  text-align: center;
}

.c0 .DayPicker_transitionContainer {
  position: relative;
  overflow: hidden;
  border-radius: 3px;
}

.c0 .DayPicker_transitionContainer__horizontal {
  -webkit-transition: height .2s ease-in-out;
  -moz-transition: height .2s ease-in-out;
  -webkit-transition: height .2s ease-in-out;
  transition: height .2s ease-in-out;
}

.c0 .DayPicker_transitionContainer__vertical {
  width: 100%;
}

.c0 .DayPicker_transitionContainer__verticalScrollable {
  padding-top: 20px;
  height: 100%;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  overflow-y: scroll;
}

.c0 .DateInput {
  margin: 0;
  padding: 0;
  background: #fff;
  position: relative;
  display: inline-block;
  width: 130px;
  vertical-align: middle;
}

.c0 .DateInput__small {
  width: 97px;
}

.c0 .DateInput__block {
  width: 100%;
}

.c0 .DateInput__disabled {
  background: #f2f2f2;
  color: #dbdbdb;
}

.c0 .DateInput_input {
  font-weight: 200;
  font-size: 19px;
  line-height: 24px;
  color: #484848;
  background-color: #fff;
  width: 100%;
  padding: 11px 11px 9px;
  border: 0;
  border-top: 0;
  border-right: 0;
  border-bottom: 2px solid transparent;
  border-left: 0;
  border-radius: 0;
}

.c0 .DateInput_input__small {
  font-size: 15px;
  line-height: 18px;
  -webkit-letter-spacing: .2px;
  -moz-letter-spacing: .2px;
  -ms-letter-spacing: .2px;
  letter-spacing: .2px;
  padding: 7px 7px 5px;
}

.c0 .DateInput_input__regular {
  font-weight: auto;
}

.c0 .DateInput_input__readOnly {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .DateInput_input__focused {
  outline: 0;
  background: #fff;
  border: 0;
  border-top: 0;
  border-right: 0;
  border-bottom: 2px solid #008489;
  border-left: 0;
}

.c0 .DateInput_input__disabled {
  background: #f2f2f2;
  font-style: italic;
}

.c0 .DateInput_screenReaderMessage {
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 .DateInput_fang {
  position: absolute;
  width: 20px;
  height: 10px;
  left: 22px;
  z-index: 2;
}

.c0 .DateInput_fangShape {
  fill: #fff;
}

.c0 .DateInput_fangStroke {
  stroke: #dbdbdb;
  fill: transparent;
}

.c0 .DateRangePickerInput {
  background-color: #fff;
  display: inline-block;
}

.c0 .DateRangePickerInput__disabled {
  background: #f2f2f2;
}

.c0 .DateRangePickerInput__withBorder {
  border-radius: 2px;
  border: 1px solid #dbdbdb;
}

.c0 .DateRangePickerInput__rtl {
  direction: rtl;
}

.c0 .DateRangePickerInput__block {
  display: block;
}

.c0 .DateRangePickerInput__showClearDates {
  padding-right: 30px;
}

.c0 .DateRangePickerInput_arrow {
  display: inline-block;
  vertical-align: middle;
  color: #484848;
}

.c0 .DateRangePickerInput_arrow_svg {
  vertical-align: middle;
  fill: #484848;
  height: 24px;
  width: 24px;
}

.c0 .DateRangePickerInput_clearDates {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  padding: 10px;
  margin: 0 10px 0 5px;
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 .DateRangePickerInput_clearDates__small {
  padding: 6px;
}

.c0 .DateRangePickerInput_clearDates_default:focus,
.c0 .DateRangePickerInput_clearDates_default:hover {
  background: #dbdbdb;
  border-radius: 50%;
}

.c0 .DateRangePickerInput_clearDates__hide {
  visibility: hidden;
}

.c0 .DateRangePickerInput_clearDates_svg {
  fill: #82888a;
  height: 12px;
  width: 15px;
  vertical-align: middle;
}

.c0 .DateRangePickerInput_clearDates_svg__small {
  height: 9px;
}

.c0 .DateRangePickerInput_calendarIcon {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  padding: 10px;
  margin: 0 5px 0 10px;
}

.c0 .DateRangePickerInput_calendarIcon_svg {
  fill: #82888a;
  height: 15px;
  width: 14px;
  vertical-align: middle;
}

.c0 .DateRangePicker {
  position: relative;
  display: inline-block;
}

.c0 .DateRangePicker__block {
  display: block;
}

.c0 .DateRangePicker_picker {
  z-index: 1;
  background-color: #fff;
  position: absolute;
}

.c0 .DateRangePicker_picker__rtl {
  direction: rtl;
}

.c0 .DateRangePicker_picker__directionLeft {
  left: 0;
}

.c0 .DateRangePicker_picker__directionRight {
  right: 0;
}

.c0 .DateRangePicker_picker__portal {
  background-color: rgba(0,0,0,.3);
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.c0 .DateRangePicker_picker__fullScreenPortal {
  background-color: #fff;
}

.c0 .DateRangePicker_closeButton {
  background: 0 0;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  padding: 15px;
  z-index: 2;
}

.c0 .DateRangePicker_closeButton:focus,
.c0 .DateRangePicker_closeButton:hover {
  color: darken(#cacccd,10%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .DateRangePicker_closeButton_svg {
  height: 15px;
  width: 15px;
  fill: #cacccd;
}

.c0 .SingleDatePickerInput {
  width: 100%;
  display: inline-block;
  background: none;
}

.c0 .SingleDatePickerInput__withBorder {
  border: none;
}

.c0 .SingleDatePickerInput_calendarIcon_svg {
  fill: #6c757d;
}

.c0 .SingleDatePicker_picker {
  background-color: #fafcff;
}

.c0 .CalendarDay__default {
  border: 1px solid #fafcff;
  color: #333333;
  background: #fafcff;
}

.c0 .CalendarDay__default:hover {
  background: #006287;
}

.c0 .CalendarDay__outside {
  border: 0;
  background: #fafcff;
  color: #333333;
}

.c0 .CalendarDay__blocked_minimum_nights {
  background: #fafcff;
  color: #6c757d;
}

.c0 .CalendarDay__blocked_minimum_nights:active,
.c0 .CalendarDay__blocked_minimum_nights:hover {
  background: #fafcff;
  color: #6c757d;
}

.c0 .CalendarDay__highlighted_calendar {
  background: #ffe8bc;
  color: #333333;
}

.c0 .CalendarDay__highlighted_calendar:active,
.c0 .CalendarDay__highlighted_calendar:hover {
  background: #ffce71;
  color: #333333;
}

.c0 .CalendarDay__selected_span {
  color: #fafcff;
}

.c0 .CalendarDay__selected_span:active,
.c0 .CalendarDay__selected_span:hover {
  background: #00ACED;
  color: #fafcff;
}

.c0 .CalendarDay__selected,
.c0 .CalendarDay__selected:active,
.c0 .CalendarDay__selected:hover {
  background: #00ACED;
  color: #fafcff;
}

.c0 .CalendarDay__blocked_calendar,
.c0 .CalendarDay__blocked_calendar:active,
.c0 .CalendarDay__blocked_calendar:hover {
  background: #006287;
  color: #6c757d;
}

.c0 .CalendarDay__blocked_out_of_range,
.c0 .CalendarDay__blocked_out_of_range:active,
.c0 .CalendarDay__blocked_out_of_range:hover {
  background: #fafcff;
  color: rgba(108,117,125,0.5);
  border: 1px solid #fafcff;
}

.c0 .CalendarDay__hovered_start_first_possible_end {
  background: #fafcff;
  border: 1px double #fafcff;
}

.c0 .CalendarDay__hovered_start_blocked_min_nights {
  background: #fafcff;
}

.c0 .CalendarMonth {
  background: #fafcff;
}

.c0 .CalendarMonth_caption {
  color: #333333;
  font-size: 18px;
  text-align: center;
  padding-top: 18px;
  padding-bottom: 37px;
  caption-side: initial;
}

.c0 .CalendarMonth_caption select {
  height: 33px;
  border-radius: 3px;
  background: #fff;
  color: #333333;
  font-size: inherit;
  padding: 0 1rem 0 1rem;
  margin: 0 .5rem 0 .5rem;
  border: 1px solid rgba(0,172,237,0.5);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0 .CalendarMonth_caption select:hover {
  border: 1px solid #006287;
}

.c0 .CalendarMonthGrid {
  background: #fafcff;
}

.c0 .DayPickerNavigation_button__default {
  border: 1px solid rgba(0,172,237,0.5);
  background-color: #fff;
  color: #757575;
}

.c0 .DayPickerNavigation_button__default:focus,
.c0 .DayPickerNavigation_button__default:hover {
  border: 1px solid #006287;
}

.c0 .DayPickerNavigation_button__default:active {
  background: #fafcff;
}

.c0 .DayPickerNavigation_button__disabled {
  cursor: default;
  border: 1px solid #fafcff;
}

.c0 .DayPickerNavigation_button__disabled:focus,
.c0 .DayPickerNavigation_button__disabled:hover {
  border: 1px solid #006287;
}

.c0 .DayPickerNavigation_button__verticalDefault {
  background: #fafcff;
}

.c0 .DayPickerNavigation_svg__horizontal {
  fill: #6c757d;
}

.c0 .DayPickerNavigation_svg__vertical {
  fill: #333333;
}

.c0 .DayPickerNavigation_svg__disabled {
  fill: #fafcff;
}

.c0 .DayPicker {
  background: #fafcff;
  position: relative;
  text-align: left;
}

.c0 .DayPicker__horizontal {
  background: #fafcff;
}

.c0 .DayPicker__withBorder {
  box-shadow: 0 3px 6px rgba(0,0,0,.05),0 0 0 1px rgba(0,0,0,.07);
  border-radius: 3px;
}

.c0 .DayPicker_weekHeader__verticalScrollable {
  border-bottom: 1px solid rgba(108,117,125,0.5);
  background: #fafcff;
}

.c0 .DateInput {
  margin: 0;
  padding: 0;
  background: none;
  position: relative;
  display: inline-block;
  width: 100%;
  vertical-align: middle;
}

.c0 .DateInput__disabled {
  background: #fafcff;
  color: rgba(108,117,125,0.5);
}

.c0 .DateInput_input {
  font-weight: 200;
  font-size: 1.4rem;
  height: 4rem;
  color: #333333;
  background-color: #fafcff;
  width: 100%;
  margin: 0;
  height: 4rem;
  background-color: #fafcff;
  border: 1px solid;
  border-color: rgba(0,172,237,0.5);
  -webkit-transition: border 0.2s linear;
  transition: border 0.2s linear;
  width: 100%;
  border-radius: 0.4rem;
}

.c0 .DateInput_input:focus,
.c0 .DateInput_input:active {
  border-color: #00ACED;
}

.c0 .DateInput_input[disabled] {
  cursor: not-allowed;
  opacity: 0.75;
}

.c0 .DateInput_input__focused {
  outline: 0;
  background: #fafcff;
}

.c0 .DateInput_input__disabled {
  background: #fafcff;
  font-style: italic;
}

.c0 .DateInput_fangShape {
  fill: #fafcff;
}

.c0 .DateInput_fangStroke {
  stroke: rgba(108,117,125,0.5);
  fill: transparent;
}

.c0 .DateRangePickerInput {
  background-color: #fafcff;
  display: inline-block;
}

.c0 .DateRangePickerInput__disabled {
  background: #fafcff;
}

.c0 .DateRangePickerInput__withBorder {
  border-radius: 3px;
  border: 1px solid rgba(108,117,125,0.5);
}

.c0 .DateRangePickerInput_arrow {
  display: inline-block;
  vertical-align: middle;
  color: #333333;
}

.c0 .DateRangePickerInput_arrow_svg {
  vertical-align: middle;
  fill: #333333;
}

.c0 .DateRangePickerInput_clearDates_default:focus,
.c0 .DateRangePickerInput_clearDates_default:hover {
  background: rgba(108,117,125,0.5);
}

.c0 .DateRangePickerInput_clearDates_svg {
  fill: #6c757d;
}

.c0 .DateRangePickerInput_calendarIcon_svg {
  fill: #6c757d;
}

.c0 .DateRangePicker_picker {
  background-color: #fafcff;
}

.c0 .DateRangePicker_picker__fullScreenPortal {
  background-color: #fafcff;
}

.c0 .DateRangePicker_closeButton:focus,
.c0 .DateRangePicker_closeButton:hover {
  color: darken(#6c757d,10%);
}

.c0 .DateRangePicker_closeButton_svg {
  fill: #6c757d;
}

<div
  class="c0"
  height="4rem"
>
  <div
    class="SingleDatePicker SingleDatePicker_1"
  >
    <div>
      <div
        class="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
      >
        <div
          class="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-date"
            aria-label="Please enter a date..."
            autocomplete="off"
            class="DateInput_input DateInput_input_1"
            id="date"
            name="date"
            placeholder="Please enter a date..."
            type="text"
            value=""
          />
          <p
            class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-date"
          >
            Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
